app/controllers/sessions_controller.rb

```diff
@@ -1,2 +1,36 @@
 class SessionsController < ApplicationController
+  before_action :assign_user, only: [:create]
+
+  def new
+  end
+
+  def create
+
+    unless @user.nil?
+      if @user.authenticate(params[:session][:password])
+        session[:user_id] = @user.id
+        redirect_to :back
+      else
+        flash[:alert] = "Invalid password."
+        render :new
+      end
+    else
+      flash[:alert] = "#{params[:session][:email]} is not a registered email address."
+      render :new
+    end
```

Comment:
Accoridng to the class style guide, we should not use unless with else, so this will need to be re-written: https://github.com/Ada-Developers-Academy/ruby-style-guide/blob/master/2-syntax.md#no-else-with-unless

We can still keep the specific messages, and you can do that with an elsif.